src/app/signalk-requests.service.ts

```diff
@@ -50,48 +82,89 @@ export class SignalkRequestsService {
       state: null,
       statusCode: null
     }
-    this.requests.push(request);  
+    this.requests.push(request);
   }
 
-  public putRequest(path: string, source: string, value: any)  {
+
+  /**
+     * Sends request to SignalK server and returns requestId.
+     * @param path SignalK full path. Automatically removes "self" if included in path.
+     * @param value Value to be sent.
+     * @param widgetUUID Optional - Subscriber's Widget UUID to be included as part of
+     * the subscribeRequest Subject response. Enables Widget specific filtering.
+     * @return requestId Identifier for this specific request. Enables Request specific filtering.
+     */
+  public putRequest(path: string, value: any, widgetUUID: string): string {
     let requestId = this.newUuid();
-    let noSelfPath = path.replace(/^(self\.)/,""); //no self in path...
+    let selfContext: string = "vessels.self";    // TODO: hard coded context. Could be dynamic at some point
```

Comment:
I'm pretty sure the spec doesn't use vessels.self anymore? Where did you see that?

Reply:
I connected with cmolete (he did the signalK-autopilot) and he help me out. It PUT request need a `vessels.self` context, but no `self.xxx` in the path. I tested and it work great.

Reply:
Oh and I just noticed, you don;t have my latest version. Next pull: I'll first integrate Widget notification to your notification framework. Easier that way for you to see where I'm going with this...